src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -258,17 +259,18 @@ class MetadataBasedItemList extends React.Component<Props, State> {
                     return cellData;
                 }
                 const { type, value, options = [] } = field;
+                const shouldShowEditIcon = isCellEditable && isString(type);
```

Comment:
Making sure data type of metadata field is valid, so that we can show edit icon and corresponding <MetadataField /> for that type.